packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -753,18 +753,33 @@ class TextSelectionOverlay {
       renderEditable: renderObject,
     ));
 
-    _handleSelectionHandleChanged(newSelection, isEnd: false);
+    _handleSelectionHandleChanged(newSelection, handleToView: TextSelectionHandleSide.baseHandle);
   }
 
   void _handleAnyDragEnd(DragEndDetails details) => _selectionOverlay.hideMagnifier(shouldShowToolbar: !_selection.isCollapsed);
 
-  void _handleSelectionHandleChanged(TextSelection newSelection, {required bool isEnd}) {
-    final TextPosition textPosition = isEnd ? newSelection.extent : newSelection.base;
+  void _handleSelectionHandleChanged(TextSelection newSelection, {required TextSelectionHandleSide handleToView}) {
+    final TextPosition positionToView;
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.iOS:
+      case TargetPlatform.macOS:
+        // On iOS and macOS, extent is always the dragged handle.
+        // (https://github.com/flutter/flutter/pull/105069)
+        positionToView = newSelection.extent;
```

Comment:
It seems surprising to me that passing handleToView won't have any effect on iOS and Mac.  Is there any other way to organize this to avoid that?

Reply:
I removed this `switch-case`.